Cubical/Algebra/CommAlgebra/UnivariatePolyList.agda

```diff
@@ -2,29 +2,218 @@
+              r ⋆ ϕ q + x · (ϕ p · ϕ q)                 ≡⟨ step7 r p ⟩
+              r ⋆ (1a · ϕ q) + (x · ϕ p) · ϕ q          ≡⟨ step8 r p ⟩
+              (r ⋆ 1a) · ϕ q + (x · ϕ p) · ϕ q          ≡⟨ sym (·DistL+ _ _ _) ⟩
+              (r ⋆ 1a + x · ϕ p) · ϕ q                  ≡⟨⟩
+              ϕ (r ∷ p) · ϕ q ∎)
+            (is-set _ _) p
+          where
+            step1 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step1 r p = inducedMap+ (r PolyConst* q) (0r ∷ (p Poly* q))
+
+            step2 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step2 r p i = ϕ (PolyConst*≡Poly* r q i) + ϕ (0r ∷ (p Poly* q))
+
+            step3 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step3 r p i = inducedMap⋆ r q i + ϕ (0r ∷ (p Poly* q))
+
+            step4 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step4 r p i = r ⋆ ϕ q + (0-actsNullifying (CommRing→Ring R) A 1a i + x · ϕ (p · q))
+
+            step5 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step5 r p i = r ⋆ ϕ q + +IdL (x · ϕ (p · q)) i
+
+            step6 : (r : ⟨ R ⟩) (p : _) → _ → _ ≡ _
+            step6 r p IH i = r ⋆ ϕ q + x · IH i
+
+            step7 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step7 r p i = r ⋆ (sym (·IdL (ϕ q)) i) + ·Assoc x (ϕ p) (ϕ q) i
+
+            step8 : (r : ⟨ R ⟩) (p : _) → _ ≡ _
+            step8 r p i = sym (⋆AssocL r 1a (ϕ q)) i + (x · ϕ p) · ϕ q
+
+      inducedHom : AlgebraHom (CommAlgebra→Algebra ListPolyCommAlgebra) A
+      fst inducedHom = inducedMap
+      snd inducedHom = makeIsAlgebraHom inducedMap1 inducedMap+ inducedMap· inducedMap⋆
+
+      inducedHomOnGenerator : fst inducedHom X ≡ x
+      inducedHomOnGenerator = inducedMapGenerator
+
+      {- Uniqueness -}
+      inducedHomUnique : (f : AlgebraHom (CommAlgebra→Algebra ListPolyCommAlgebra) A)
+                         → fst f X ≡ x
+                         → f ≡ inducedHom
+      inducedHomUnique f fX≡x =
+        Σ≡Prop
+          (λ _ → isPropIsAlgebraHom _ _ _ _)
+          λ i p → pwEq p i
+        where
+        open IsAlgebraHom (snd f)
+        pwEq : (p : ⟨ ListPolyCommAlgebra ⟩) → fst f p ≡ fst inducedHom p
+        pwEq =
+          ElimProp R
+            (λ p → fst f p ≡ fst inducedHom p)
+            pres0
+            (λ r p IH →
+              fst f (r ∷ p)                    ≡[ i ]⟨ fst f ((sym (+IdR r) i) ∷ sym (Poly+Lid p) i) ⟩
+              fst f ([ r ] + (0r ∷ p))         ≡[ i ]⟨ fst f ([ useSolver r i ] + sym (X*Poly p) i) ⟩
+              fst f (r ⋆ 1a + X · p)           ≡⟨ pres+ (r ⋆ 1a) (X · p) ⟩
+              fst f (r ⋆ 1a) + fst f (X · p)   ≡[ i ]⟨ pres⋆ r 1a i + pres· X p i ⟩
+              r ⋆ fst f 1a + fst f X · fst f p ≡[ i ]⟨ r ⋆ pres1 i + fX≡x i · fst f p ⟩
+              r ⋆ 1a + x · fst f p             ≡[ i ]⟨ r ⋆ 1a + (x · IH i) ⟩
+              r ⋆ 1a + x · inducedMap p        ≡⟨⟩
+              inducedMap (r ∷ p) ∎)
+            (is-set _ _)
+          where
+            useSolver : (r : ⟨ R ⟩) → r ≡ (r · 1r) + 0r
+            useSolver = solve R
```

Comment:
Would it be hard to state the universal property as "there is an equivalence between A and the R-homs from R[X] to A"?
I feel like that would look nice if you write it down explicitly.

Reply:
Yes, good idea. It shouldn't be hard, I'll try to that later...
Btw.: Have you thought about a general definition of UMP for algebraic things, that would allow us to change between the different forms and have some simple consequences? I would have some use for that, but haven't tried to write something. I only tried to use category theory (from the library), but that is not general enough (because of universe levels).

Reply:
That would be cool. What do the textbooks say, do they ever give a general definition?
Maybe it helps to think about typical consequences, like the only homomorphism from the object to itself is the identity etc...

Reply:
Yes, that's exactly something I would have liked to use a couple of times. That already seems to need functoriality, though, meaning we would have to redefine that with varying universe levels.

Reply:
Made an issue for that: #943

Reply:
Reformulated this universal property like suggested.